core/src/Streamly/Internal/Data/Stream/StreamD/Nesting.hs

```diff
@@ -1896,16 +1897,16 @@ data GroupByState st fs a b
     | GroupingYield !b (GroupByState st fs a b)
     | GroupingDone
 
-{-# INLINE_NORMAL groupsBy #-}
-groupsBy :: Monad m
+{-# INLINE_NORMAL groupsWhile #-}
+groupsWhile :: Monad m
```

Comment:
Add a comment on this that the argument order is different from groupsBy. otherwise users who replace groupsBy with this will run into problems.